Adding Themable mixin and parts

manolo opened the pull request:
Fixes #462

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/vaadin-combo-box/468)
<!-- Reviewable:end -->

tomivirkki commented:
Review status: 0 of 8 files reviewed at latest revision, 9 unresolved discussions.

---

*[vaadin-combo-box-overlay.html, line 45 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Kni3eZQvunW1SIYPLYU:-Kni3eZQvunW1SIYPLYV:b-yy4qfg) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-overlay.html#L45)):*
> ```HTML
>           }
>         </style>
>         <vaadin-spinner part="spinner" active="[[loading]]"></vaadin-spinner>
> ```

This part isn't needed. The spinner extends themable mixin by itself.

---

*[vaadin-combo-box-overlay.html, line 46 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Kni47CSib74-b3nQ3v2:-Kni47CSib74-b3nQ3v3:bqa3nnz) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-overlay.html#L46)):*
> ```HTML
>         </style>
>         <vaadin-spinner part="spinner" active="[[loading]]"></vaadin-spinner>
>         <div part="scroller" id="scroller" on-tap="_stopPropagation" on-touchstart="_onTouchStart" on-touchend="_preventDefault"
> ```

Don't make this a part (we don't want to support these API's once the contents are replaced by a vaadin-list-box)

---

*[vaadin-combo-box-overlay.html, line 48 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Kni4gMFwg4QHgqr7HgE:-Kni4gMFwg4QHgqr7HgF:bqa3nnz) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-overlay.html#L48)):*
> ```HTML
>         <div part="scroller" id="scroller" on-tap="_stopPropagation" on-touchstart="_onTouchStart" on-touchend="_preventDefault"
>           on-scroll="_onScroll" hidden$="[[loading]]">
>           <iron-list part="selector" id="selector" role="listbox" items="[[_items]]" scroll-target="[[_scroller]]">
> ```

Don't make this a part (we don't want to support these API's once the contents are replaced by a vaadin-list-box)

---

*[vaadin-combo-box-styles.html, line 47 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KnhzNpxYkL5c05iNeyM:-KnhzNpxYkL5c05iNeyN:bhmdxwv) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-styles.html#L47)):*
> ```HTML
>         cursor: pointer;
>         transition: color 0.2s;
>         height: 24px;
> ```

The buttons should not be increased: https://www.dropbox.com/s/3c3qs2evy8jnfhv/Screenshot%202017-06-28%2012.13.54.png?dl=0

---

*[vaadin-combo-box-styles.html, line 95 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Kni6MuDGjJq2ekT0Un1:-Kni6MuDGjJq2ekT0Un2:bcop8w6) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-styles.html#L95)):*
> ```HTML
>       }
> 
>       :host([hidden]) {
> ```

This and the default display feel more like functional styles (should be in the element)

---

*[vaadin-combo-box-styles.html, line 107 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Kni6tIp9ouYTg6tMORr:-Kni6tIp9ouYTg6tMORs:bg22683) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-styles.html#L107)):*
> ```HTML
> </dom-module>
> 
> <dom-module id="vaadin-spinner-default-theme" theme-for="vaadin-spinner">
> ```

The spinner is never visible

---

*[vaadin-combo-box.html, line 21 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Kni4zAmn_-KbEJwrDaF:-Kni4zAmn_-KbEJwrDaG:bke81zm) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box.html#L21)):*
> ```HTML
>       :host {
>         display: inline-block;
>         padding: 8px 0;
> ```

Probably not a functional style (could be moved to default theme modules)

---

*[vaadin-spinner.html, line 14 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Kni0nPG8l1VVxZKgJVm:-Kni0nPG8l1VVxZKgJVn:b-7ki3re) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-spinner.html#L14)):*
> ```HTML
>     <style>
>       /* Anim */
>       @keyframes vaadin-spin-360 {
> ```

Not directly related to this pr but could you please rename/scope this to "vaadin-combo-box-spinner-spin". "vaadin-spin-360" is too ambiguous. P1+shady scopes animation names automatically but AFAIK this is not the case with P2+shady.

---

*[vaadin-spinner.html, line 42 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Kni0Ngkbg41opry1GA7:-Kni0Ngkbg41opry1GA8:b3nb280) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-spinner.html#L42)):*
> ```HTML
>        * @mixes Vaadin.ThemableMixin
>        */
>       class SpinnerElement extends Vaadin.ThemableMixin(Polymer.Element) {
> ```

This needs to be renamed to "vaadin-combo-box-spinner" since it's not a shared element (it's strange if theme-for="vaadin-spinner" only affects combo-box's spinner. Also, if some other vaadin-element defines its own "vaadin-spinner", there will be a clash).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 8 files reviewed at latest revision, 9 unresolved discussions.

---

*[vaadin-combo-box-styles.html, line 47 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KnhzNpxYkL5c05iNeyM:-Kni9Sta6h7KcZTRxyfJ:b4y0lmq) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-styles.html#L47)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

The buttons should not be increased: https://www.dropbox.com/s/3c3qs2evy8jnfhv/Screenshot%202017-06-28%2012.13.54.png?dl=0
</blockquote></details>

I think they should based on comments of Anton in another PR who suggested to be big enough for mobile.
The problem here is how to target the height of the input so as it is the same than the buttons when they are removed, it seems it's not doable without exposing something else in the text-field

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[vaadin-combo-box-styles.html, line 47 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KnhzNpxYkL5c05iNeyM:-KniDTxIodl5ce-Gi4XX:b-bgn9jv) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-styles.html#L47)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

I think they should based on comments of Anton in another PR who suggested to be big enough for mobile.
The problem here is how to target the height of the input so as it is the same than the buttons when they are removed, it seems it's not doable without exposing something else in the text-field
</blockquote></details>

You should be able to increase the `::before` element (probably need to make is absolute) without increasing the button itself so the component layout isn't affected.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

jouni commented:
Review status: 0 of 8 files reviewed at latest revision, 9 unresolved discussions.

---

*[vaadin-combo-box-styles.html, line 47 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KnhzNpxYkL5c05iNeyM:-KniYrGLPcSkRW7Jq7Lh:b-pnwd8x) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-styles.html#L47)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

You should be able to increase the `::before` element (probably need to make is absolute) without increasing the button itself so the component layout isn't affected.
</blockquote></details>

I think the default styles don’t necessarily need to worry about touch friendliness. That responsibility is mainly on the actual theme implementations. But if we can agree on a common approach that works for our themes, I think we can use that.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

jouni commented:
Review status: 0 of 8 files reviewed at latest revision, 9 unresolved discussions.

---

*[vaadin-combo-box-styles.html, line 47 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KnhzNpxYkL5c05iNeyM:-KnieO-rSrCujdMxseQr:bs4ncze) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-styles.html#L47)):*
<details><summary><i>Previously, jouni (Jouni Koivuviita) wrote…</i></summary><blockquote>

I think the default styles don’t necessarily need to worry about touch friendliness. That responsibility is mainly on the actual theme implementations. But if we can agree on a common approach that works for our themes, I think we can use that.
</blockquote></details>

I guess negative margins can be used to avoid increasing the input height.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 8 files reviewed at latest revision, 9 unresolved discussions.

---

*[vaadin-combo-box-overlay.html, line 45 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Kni3eZQvunW1SIYPLYU:-Knmfnq2Vzz_HcGwq0wP:b-896fix) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-overlay.html#L45)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

This part isn't needed. The spinner extends themable mixin by itself.
</blockquote></details>

Done.

---

*[vaadin-combo-box-overlay.html, line 46 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Kni47CSib74-b3nQ3v2:-KnmfwmkboJnGPMBJeMe:b-896fix) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-overlay.html#L46)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Don't make this a part (we don't want to support these API's once the contents are replaced by a vaadin-list-box)
</blockquote></details>

Done.

---

*[vaadin-combo-box-overlay.html, line 48 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Kni4gMFwg4QHgqr7HgE:-Knmg6P-6CLVmmH7eYXc:b-896fix) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-overlay.html#L48)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Don't make this a part (we don't want to support these API's once the contents are replaced by a vaadin-list-box)
</blockquote></details>

Done.

---

*[vaadin-combo-box-styles.html, line 47 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KnhzNpxYkL5c05iNeyM:-KnmuFOCjLDa9jjvw001:b2t6omn) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-styles.html#L47)):*
<details><summary><i>Previously, jouni (Jouni Koivuviita) wrote…</i></summary><blockquote>

I guess negative margins can be used to avoid increasing the input height.
</blockquote></details>

This is not an issue in this PR but in the text-field that should define a default height for the `part=input-field`.
I have opened an issue:  https://github.com/vaadin/vaadin-text-field/issues/99

I think we should leave 24px size for icons here to be aligned with date-picker default sizes and to apply @platosha  suggestion that icons should be clickable in mobile devices.

---

*[vaadin-combo-box-styles.html, line 95 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Kni6MuDGjJq2ekT0Un1:-Knmw4LkgVMVuBTpqvNp:b-896fix) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-styles.html#L95)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

This and the default display feel more like functional styles (should be in the element)
</blockquote></details>

Done.

---

*[vaadin-combo-box.html, line 21 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Kni4zAmn_-KbEJwrDaF:-Knmy6UGEapJdBJp_TDt:b-896fix) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box.html#L21)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Probably not a functional style (could be moved to default theme modules)
</blockquote></details>

Done.

---

*[vaadin-spinner.html, line 14 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Kni0nPG8l1VVxZKgJVm:-Knn26Roh_HsICvfULft:b-896fix) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-spinner.html#L14)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Not directly related to this pr but could you please rename/scope this to "vaadin-combo-box-spinner-spin". "vaadin-spin-360" is too ambiguous. P1+shady scopes animation names automatically but AFAIK this is not the case with P2+shady.
</blockquote></details>

Done.

---

*[vaadin-spinner.html, line 42 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Kni0Ngkbg41opry1GA7:-Knn26xaafQEhHbcPUxN:b-896fix) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-spinner.html#L42)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

This needs to be renamed to "vaadin-combo-box-spinner" since it's not a shared element (it's strange if theme-for="vaadin-spinner" only affects combo-box's spinner. Also, if some other vaadin-element defines its own "vaadin-spinner", there will be a clash).
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

jouni commented:
Just wondering, should we eventually use vaadin-progress instead of vaadin-combo-box-spinner?

manolo commented:
Sure we should use it, but not in this PR

---

Review status: 0 of 11 files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[vaadin-combo-box-styles.html, line 107 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Kni6tIp9ouYTg6tMORr:-KnnFLEUlZSQWl4tUc52:b-896fix) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-styles.html#L107)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

The spinner is never visible
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-:-KnnFBGTezE4kzw7qRiJ:bs56pym)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
Probably yes, so "vaadin-combo-box-spinner" should not have any themable parts at this point so we can switch it later.

---

Review status: 0 of 8 files reviewed at latest revision, 5 unresolved discussions.

---

*[vaadin-combo-box-styles.html, line 71 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KnnnuX7VJQFX8HAmRVC:-KnnnuX7VJQFX8HAmRVD:b-2ypra4) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/9e54f221627530c2767ce856b4af3c26fffa5215/vaadin-combo-box-styles.html#L71)):*
> ```HTML
> 
>       [part="toggle-button"] {
>         transition: all 0.2s !important;
> ```

Default theme should not have any transitions. Also all the color changes should be dropped (like in this PR https://github.com/vaadin/vaadin-date-picker/pull/393)

---

*[vaadin-combo-box-styles.html, line 110 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Knno6TjI54ngLAMKoJ_:-Knno6TjI54ngLAMKoJa:b-eyds78) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/9e54f221627530c2767ce856b4af3c26fffa5215/vaadin-combo-box-styles.html#L110)):*
> ```HTML
>     <style>
>       :host {
>         border: 2px solid var(--primary-color, #03A9F4);
> ```

Drop `--primary-color`

---

*[vaadin-spinner.html, line 42 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Kni0Ngkbg41opry1GA7:-KnnnjJC0u0JS81F8bLz:b2oqdmz) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-spinner.html#L42)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

It's not committed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-:-KnnnTMHmHrFrR0kBEUZ:b-wm685z)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 8 files reviewed at latest revision, 5 unresolved discussions.

---

*[vaadin-combo-box-styles.html, line 71 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KnnnuX7VJQFX8HAmRVC:-Knq4YsnXwGazWw3bLVC:b-7tr391) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/9e54f221627530c2767ce856b4af3c26fffa5215/vaadin-combo-box-styles.html#L71)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Default theme should not have any transitions. Also all the color changes should be dropped (like in this PR https://github.com/vaadin/vaadin-date-picker/pull/393)
</blockquote></details>

Dizzy here, first @saulis  ask for adding color, and now you for removing them.
The simplification in the date-picker happened after this PR, so this was based in the original date-picker.
Probably all those adjustments might be done in a separated PR.

Anyways simplified styles, hopefully it's now Ok

---

*[vaadin-combo-box-styles.html, line 110 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Knno6TjI54ngLAMKoJ_:-Knq47naln3hnbnS26wZ:b-896fix) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/9e54f221627530c2767ce856b4af3c26fffa5215/vaadin-combo-box-styles.html#L110)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Drop `--primary-color`
</blockquote></details>

Done.

---

*[vaadin-spinner.html, line 42 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Kni0Ngkbg41opry1GA7:-Knq3dNxRJ6gpGznzTds:b-896fix) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-spinner.html#L42)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

It's not committed
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

jouni commented:
> Dizzy here, first @Saulis ask for adding color, and now you for removing them.
The simplification in the date-picker happened after this PR, so this was based in the original date-picker.
Probably all those adjustments might be done in a separated PR.

Yeah, sorry about that. I didn’t pay attention to the date picker default styles too closely before, so some uneanted styles slipped through.

tomivirkki commented:
Review status: 0 of 9 files reviewed at latest revision, 5 unresolved discussions.

---

*[vaadin-combo-box-styles.html, line 47 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KnhzNpxYkL5c05iNeyM:-KnsA1kaCV7hD7lLIGrV:bojdys1) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-styles.html#L47)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

This is not an issue in this PR but in the text-field that should define a default height for the `part=input-field`.
I have opened an issue:  https://github.com/vaadin/vaadin-text-field/issues/99

I think we should leave 24px size for icons here to be aligned with date-picker default sizes and to apply @platosha  suggestion that icons should be clickable in mobile devices.
</blockquote></details>

Not sure what you guys agreed yesterday but shouldn't the fixed `font-size: 16px` be dropped also?

---

*[vaadin-combo-box-styles.html, line 85 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KnsAaW6yNsQcunQHuYd:-KnsAaW6yNsQcunQHuYe:bqxrjgq) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-styles.html#L85)):*
> ```HTML
>         cursor: pointer;
>         padding: 13px 16px;
>         color: #212121;
> ```

Why this color as default?

---

*[vaadin-combo-box-styles.html, line 92 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KnsAjf70wGp2DxPqJNz:-KnsAjf70wGp2DxPqJO-:bljybno) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-styles.html#L92)):*
> ```HTML
> 
>       :host([selected]) {
>         color: #3f51b5;
> ```

I'm pretty sure we don't want "--paper-indigo-500" as the default selected color

---

*[vaadin-combo-box-styles.html, line 107 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Kni6tIp9ouYTg6tMORr:-KnsAKMzI713Q2R2AyNY:bjnbgid) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-styles.html#L107)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

I still don't see the spinner.

---

*[vaadin-combo-box-styles.html, line 110 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Knno6TjI54ngLAMKoJ_:-Kns8l_nbOtIzQtZcDeD:b-2igjt6) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/9e54f221627530c2767ce856b4af3c26fffa5215/vaadin-combo-box-styles.html#L110)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

The color is still something from paper-styles (--paper-light-blue-500). Should be a neutral color.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 9 files reviewed at latest revision, 5 unresolved discussions.

---

*[vaadin-combo-box-styles.html, line 110 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Knno6TjI54ngLAMKoJ_:-KnsBlTGyHwUrhPqF1D9:bugcqsm) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/9e54f221627530c2767ce856b4af3c26fffa5215/vaadin-combo-box-styles.html#L110)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

The color is still something from paper-styles (--paper-light-blue-500). Should be a neutral color.
</blockquote></details>

Please can you provide a color you are ok with ? this was the previous one. What is a neutral color ? 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[vaadin-combo-box-styles.html, line 110 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Knno6TjI54ngLAMKoJ_:-KnsCi6_WiLEeUxyle6g:bfaiicf) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/9e54f221627530c2767ce856b4af3c26fffa5215/vaadin-combo-box-styles.html#L110)):*
It's not my call. You can look for inspiration from other elements or ask el designer.

>this was the previous one

Yes, the previous one was material

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 9 files reviewed at latest revision, 5 unresolved discussions.

---

*[vaadin-combo-box-styles.html, line 85 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KnsAaW6yNsQcunQHuYd:-KnsCQ_i2A1-xaJhzxz3:basew2k) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-styles.html#L85)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Why this color as default?
</blockquote></details>

any suggestion ?

---

*[vaadin-combo-box-styles.html, line 110 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Knno6TjI54ngLAMKoJ_:-KnsDf5g9NmF9_VvRpWP:b-39zo3t) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/9e54f221627530c2767ce856b4af3c26fffa5215/vaadin-combo-box-styles.html#L110)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

It's not my call. You can look for inspiration from other elements or ask el designer.

>this was the previous one

Yes, the previous one was material
</blockquote></details>

Is this a blocker ? 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
Review status: 0 of 9 files reviewed at latest revision, 5 unresolved discussions.

---

*[vaadin-combo-box-styles.html, line 85 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KnsAaW6yNsQcunQHuYd:-KnsEx9KgCuI3e282SW_:b2qgxdx) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-styles.html#L85)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

any suggestion ?
</blockquote></details>

I suggest not having a fixed default text color at all and let the themes decide

---

*[vaadin-combo-box-styles.html, line 110 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Knno6TjI54ngLAMKoJ_:-KnsE8BmklV3Isa3Z1eC:bfp2pqo) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/9e54f221627530c2767ce856b4af3c26fffa5215/vaadin-combo-box-styles.html#L110)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Is this a blocker ? 
</blockquote></details>

Having paper-styles colors in the default theme? I think so. WDYT?

The default appearance of all the other vaadin elements are neutral with no colors. This would be a weird exception. Why would combo-box not be aligned?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 9 files reviewed at latest revision, 5 unresolved discussions.

---

*[vaadin-combo-box-styles.html, line 110 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Knno6TjI54ngLAMKoJ_:-KnsIn5_Z6SlYGrzWTW7:b-uw3z4z) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/9e54f221627530c2767ce856b4af3c26fffa5215/vaadin-combo-box-styles.html#L110)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Having paper-styles colors in the default theme? I think so. WDYT?

The default appearance of all the other vaadin elements are neutral with no colors. This would be a weird exception. Why would combo-box not be aligned?
</blockquote></details>

This is the color we had before, I try to change as few things as possible in my PR.
There is no a clear design about what would be the default style for the combo box items, so for this PR should be ok to maintain the previous visual and cover in a separated PR those changes once they are designed. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[vaadin-combo-box-styles.html, line 110 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Knno6TjI54ngLAMKoJ_:-KnsM4Aq5ebqNM4lUCW-:bwok6k0) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/9e54f221627530c2767ce856b4af3c26fffa5215/vaadin-combo-box-styles.html#L110)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

This is the color we had before, I try to change as few things as possible in my PR.
There is no a clear design about what would be the default style for the combo box items, so for this PR should be ok to maintain the previous visual and cover in a separated PR those changes once they are designed. 
</blockquote></details>

The whole purpose of the ticket is to change from paper elements to vaadin-elements, material-design to neutral. This is what we've done with all the other compnents. It's not that complicated to change this color code that it needs a dedicated PR.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 9 files reviewed at latest revision, 5 unresolved discussions.

---

*[vaadin-combo-box-styles.html, line 107 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Kni6tIp9ouYTg6tMORr:-KnsNkR3Ff7pu3SQm-dK:b41hmfd) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-styles.html#L107)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

I still don't see the spinner.
</blockquote></details>

Forgot to rebase with the other PR where it was fixed. Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 9 files reviewed at latest revision, 5 unresolved discussions.

---

*[vaadin-combo-box-styles.html, line 47 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KnhzNpxYkL5c05iNeyM:-KnsOG6nYi8CCz_OsKu2:b-896fix) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-styles.html#L47)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Not sure what you guys agreed yesterday but shouldn't the fixed `font-size: 16px` be dropped also?
</blockquote></details>

Done.

---

*[vaadin-combo-box-styles.html, line 85 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KnsAaW6yNsQcunQHuYd:-KnsORoPvoQARcX0ZT-8:b-896fix) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-styles.html#L85)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

I suggest not having a fixed default text color at all and let the themes decide
</blockquote></details>

Done.

---

*[vaadin-combo-box-styles.html, line 92 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KnsAjf70wGp2DxPqJNz:-KnsOfxKpkAIZ_CVBXsq:b-896fix) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/8e04709e8ebe6b7d294a4a3df8c1131de9835626/vaadin-combo-box-styles.html#L92)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

I'm pretty sure we don't want "--paper-indigo-500" as the default selected color
</blockquote></details>

Done.

---

*[vaadin-combo-box-styles.html, line 110 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Knno6TjI54ngLAMKoJ_:-KnsPE8nHBMNLLtY5J7a:b-896fix) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/9e54f221627530c2767ce856b4af3c26fffa5215/vaadin-combo-box-styles.html#L110)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

The whole purpose of the ticket is to change from paper elements to vaadin-elements, material-design to neutral. This is what we've done with all the other compnents. It's not that complicated to change this color code that it needs a dedicated PR.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

jouni commented:
*[vaadin-combo-box-styles.html, line 110 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Knno6TjI54ngLAMKoJ_:-Knsak6dass1c1h2lFSa:b6txv1s) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/9e54f221627530c2767ce856b4af3c26fffa5215/vaadin-combo-box-styles.html#L110)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

You can blame me for this misunderstanding. I didn’t provide any guidance for the default styles beforehand, and now it’s causing a lot of back-and-forth discussion.

A couple of general guidelines for the future, to help in deciding what to include in the default styles:

- If a style is something that at least 80% of themes will add, it’s a good idea to have that in the default styles already. These usually are related to layout and positioning.
- The same applies for selectors as well, so try to avoid complex selectors if it is expected that more than 20% of the themes will want to override the styles for that selectors.
- In general, the default styles should mainly be about making the element functional, as in all features being presented and working, but not necessarily optimized for usability or visual appearance.
- Purely visual styles, such as box-shadows or color, are okay to add when it’s expected that most themes will either add or override those in any case. Just make sure they are easy to override (no complex selectors).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

jouni commented:
Not sure if it’s in the scope of this PR, but the “Item template” demos are all pretty broken at the moment.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-:-KnsfSvUam6y5EfyoJ_o:b815e3i)*
<!-- Sent from Reviewable.io -->

jouni commented:
*[vaadin-combo-box-styles.html, line 50 at r4](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Knsg6US7w7SOh8TQoLg:-Knsg6US7w7SOh8TQoLh:b-urz3q7) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/2b2cdecf6a4b11a1b470984d4d0ac9bb1aa10995/vaadin-combo-box-styles.html#L50)):*
> ```HTML
>         -webkit-transform: rotate(180deg);
>         transform: rotate(180deg);
>       }
> ```

This should be removed from the default styles, not all themes will want to do this.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

jouni commented:
*[vaadin-combo-box.html, line 53 at r4](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KnsgRzOKv6_nfWONM-A:-KnsgRzPGPnRp164Vsza:bpb2hcl) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/2b2cdecf6a4b11a1b470984d4d0ac9bb1aa10995/vaadin-combo-box.html#L53)):*
> ```HTML
>       :host([readonly]) [part="toggle-button"],
>       :host(:not([has-value])) [part="clear-button"] {
>         display: none;
> ```

As discussed yesterday in Flowdock, these styles should be removed. We should not hide the buttons for disabled or readonly states. The opacity of the input-field part could be reduced when disabled.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

jouni commented:
*[vaadin-combo-box.html, line 53 at r4](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KnsgRzOKv6_nfWONM-A:-KnsuY36XmM-3IrQXtSW:b1ygwh0) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/2b2cdecf6a4b11a1b470984d4d0ac9bb1aa10995/vaadin-combo-box.html#L53)):*
<details><summary><i>Previously, jouni (Jouni Koivuviita) wrote…</i></summary><blockquote>

As discussed yesterday in Flowdock, these styles should be removed. We should not hide the buttons for disabled or readonly states. The opacity of the input-field part could be reduced when disabled.
</blockquote></details>

Oh, correction: the toggle button should not be hidden. The clear button should be hidden for disabled and readonly states.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 9 files reviewed at latest revision, 7 unresolved discussions.

---

*[vaadin-combo-box-styles.html, line 50 at r4](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-Knsg6US7w7SOh8TQoLg:-Knx7BWjB57Lnyf2thEM:b-896fix) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/2b2cdecf6a4b11a1b470984d4d0ac9bb1aa10995/vaadin-combo-box-styles.html#L50)):*
<details><summary><i>Previously, jouni (Jouni Koivuviita) wrote…</i></summary><blockquote>

This should be removed from the default styles, not all themes will want to do this.
</blockquote></details>

Done.

---

*[vaadin-combo-box.html, line 53 at r4](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KnsgRzOKv6_nfWONM-A:-Knx95jOHL03Egf6oio7:b-jtyvdr) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/2b2cdecf6a4b11a1b470984d4d0ac9bb1aa10995/vaadin-combo-box.html#L53)):*
<details><summary><i>Previously, jouni (Jouni Koivuviita) wrote…</i></summary><blockquote>

Oh, correction: the toggle button should not be hidden. The clear button should be hidden for disabled and readonly states.
</blockquote></details>

Opacity is already handled by text-field, so no adding that rule, not hiding arrow for any state.
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

manolo commented:
> Not sure if it’s in the scope of this PR, but the “Item template” demos are all pretty broken at the moment.

There was a bug in code manifested when moved styles to the themable item element. Default styles and injected theme blocks were removed when there was a template. Fixed

---

Review status: 0 of 12 files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-:-Knxptj8tNMV5l21urnp:ba2g3ma)*
<!-- Sent from Reviewable.io -->

Toshegg commented:
Review status: 0 of 15 files reviewed at latest revision, 10 unresolved discussions.

---

*[index.html, line 5 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KoMMt0Cw7KpJrsvnG1f:-KoMMt0Cw7KpJrsvnG1g:b-ezhxex) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/cf40b0a174c0b8de2e41f61359408e65a5f35c91/index.html#L5)):*
> ```HTML
> <html>
>   <head>
>     <title>vaadin-split-layout</title>
> ```

Isn't it combo-box, but not split-layout?

---

*[vaadin-combo-box.html, line 86 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KoMO0OHShsiuEbS_b2g:-KoMO0OHShsiuEbS_b2h:b7kgkr1) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/cf40b0a174c0b8de2e41f61359408e65a5f35c91/vaadin-combo-box.html#L86)):*
> ```HTML
>     </vaadin-text-field>
> 
>     <!-- Cannot use part since the internal vaadin-overlay is dettached. Theme vaadin-overlay instead -->
> ```

Typo: detached.

---

*[vaadin-overlay.html, line 11 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KoMOCzfcoeV44I_-Is5:-KoMOCzfcoeV44I_-Is6:b-x1wc91) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/cf40b0a174c0b8de2e41f61359408e65a5f35c91/vaadin-overlay.html#L11)):*
> ```HTML
> <link rel="import" href="../vaadin-themable-mixin/vaadin-themable-mixin.html">
> 
> <!-- TODO: remove this element from combo-box and use shared vaadin-overlay insttead -->
> ```

Typo: instead.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 15 files reviewed at latest revision, 10 unresolved discussions.

---

*[index.html, line 5 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KoMMt0Cw7KpJrsvnG1f:-KoMenhZOQabeHzC9UdS:b-896fix) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/cf40b0a174c0b8de2e41f61359408e65a5f35c91/index.html#L5)):*
<details><summary><i>Previously, Toshegg wrote…</i></summary><blockquote>

Isn't it combo-box, but not split-layout?
</blockquote></details>

Done.

---

*[vaadin-combo-box.html, line 86 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KoMO0OHShsiuEbS_b2g:-KoMexHh2_APqYYyos9f:b-896fix) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/cf40b0a174c0b8de2e41f61359408e65a5f35c91/vaadin-combo-box.html#L86)):*
<details><summary><i>Previously, Toshegg wrote…</i></summary><blockquote>

Typo: detached.
</blockquote></details>

Done.

---

*[vaadin-overlay.html, line 11 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KoMOCzfcoeV44I_-Is5:-KoMeh0_1Lb1AUEim3YZ:b-896fix) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/cf40b0a174c0b8de2e41f61359408e65a5f35c91/vaadin-overlay.html#L11)):*
<details><summary><i>Previously, Toshegg wrote…</i></summary><blockquote>

Typo: instead.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

Toshegg commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 8 files at r1, 1 of 5 files at r2, 2 of 3 files at r3, 8 of 11 files at r6, 3 of 3 files at r7.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-:-KoMf_2GYFpDg1idPsUC:bnfp4nl)*
<!-- Sent from Reviewable.io -->

platosha commented:
Review status: all files reviewed at latest revision, 11 unresolved discussions.

---

*[vaadin-combo-box-item.html, line 14 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KoMvYXxAcyPux2lRjb8:-KoMvYXyLuMthnLJ4I1m:b-o9bonw) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/cf40b0a174c0b8de2e41f61359408e65a5f35c91/vaadin-combo-box-item.html#L14)):*
> ```HTML
>       :host {
>         display: block;
>         overflow: hidden;
> ```

Is it necessary? Maybe leave a comment why?

---

*[vaadin-combo-box-spinner.html, line 25 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KoN2n-0QIsF0tsiNQZh:-KoN2n-0QIsF0tsiNQZi:bimxdfh) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/cf40b0a174c0b8de2e41f61359408e65a5f35c91/vaadin-combo-box-spinner.html#L25)):*
> ```HTML
>         box-sizing: border-box;
>         left: 50%;
>         top: 50%;
> ```

BTW centering with `left: 50%; top: 50%;` is incomplete without `translate: transform(-50%, -50%);`

(fixing this is unnecessary, I would rather remove the themable spinner altogether, as commented below)

---

*[vaadin-combo-box-spinner.html, line 42 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KoN1qyZ9vdGcywg8H6p:-KoN1qyZ9vdGcywg8H6q:b-g9gcrb) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/cf40b0a174c0b8de2e41f61359408e65a5f35c91/vaadin-combo-box-spinner.html#L42)):*
> ```HTML
>        * @mixes Vaadin.ThemableMixin
>        */
>       class ComboBoxSpinnerElement extends Vaadin.ThemableMixin(Polymer.Element) {
> ```

Do we really need a separate themable **combo-box** spinner?

It complicates theming of the combo-box.

Should rather reuse undefined `<vaadin-progress-circle>` here in future. As of now, I think, it’s fine to proceed with non-themable spinner for a while.

---

*[demo/item-template.html, line 140 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KoN2WNgg7lV7lKmtQvJ:-KoN2WNgg7lV7lKmtQvK:bbvwuj1) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/cf40b0a174c0b8de2e41f61359408e65a5f35c91/demo/item-template.html#L140)):*
> ```HTML
>           <template>
>             <paper-icon-item>
>               <img src="[[item.picture.thumbnail]]" style="border-radius: 50%; width: 48px; height: 48px; margin: 5px 20px 5px -40px;" item-icon>
> ```

Don’t do this, please.

Use `slot="item-icon"` instead of hacking the margins.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: all files reviewed at latest revision, 11 unresolved discussions.

---

*[vaadin-combo-box-spinner.html, line 25 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KoN2n-0QIsF0tsiNQZh:-KoN7IcaIjmsd8H-QrHI:b-m5akoi) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/cf40b0a174c0b8de2e41f61359408e65a5f35c91/vaadin-combo-box-spinner.html#L25)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

BTW centering with `left: 50%; top: 50%;` is incomplete without `translate: transform(-50%, -50%);`

(fixing this is unnecessary, I would rather remove the themable spinner altogether, as commented below)
</blockquote></details>

That is copied from the original element. Since in vaadin-progress it should be fixed, I would transfer the resolution of this to the PR replacing the temporary spinner here.

---

*[vaadin-combo-box-spinner.html, line 42 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KoN1qyZ9vdGcywg8H6p:-KoN6C5y4Gi7sTnAkddO:b-60q8uz) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/cf40b0a174c0b8de2e41f61359408e65a5f35c91/vaadin-combo-box-spinner.html#L42)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Do we really need a separate themable **combo-box** spinner?

It complicates theming of the combo-box.

Should rather reuse undefined `<vaadin-progress-circle>` here in future. As of now, I think, it’s fine to proceed with non-themable spinner for a while.
</blockquote></details>

It's here until we replace with the new element. This PR was much before than the new component was started.
Let's do it in a separate PR.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

manolo commented:
Rebased

---

Review status: 9 of 51 files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[vaadin-combo-box-item.html, line 14 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KoMvYXxAcyPux2lRjb8:-KoW4F9_rVVrpefQN_iv:b-sfdxwt) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/cf40b0a174c0b8de2e41f61359408e65a5f35c91/vaadin-combo-box-item.html#L14)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Is it necessary? Maybe leave a comment why?
</blockquote></details>

Left by error. Not needed. Done.

---

*[demo/item-template.html, line 140 at r6](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KoN2WNgg7lV7lKmtQvJ:-KoW4ekSfW38l_E85KUQ:bbjhwug) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/cf40b0a174c0b8de2e41f61359408e65a5f35c91/demo/item-template.html#L140)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Don’t do this, please.

Use `slot="item-icon"` instead of hacking the margins.
</blockquote></details>

Ups, didn't realise the item-icon attribute there. Changed to slot as suggested. Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-:-KoW6-CfiuGQFg6B92C-:b9x2p3v)*
<!-- Sent from Reviewable.io -->

platosha commented:
Reviewed 8 of 8 files at r8.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[bower.json, line 28 at r8](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KofYhq0K8d5AY_nngml:-KofYhq0K8d5AY_nngmm:biig38g) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/c7ec7f3d61e27554ef03558d60f94ed74e3d9484/bower.json#L28)):*
> ```JSON
>     "vaadin-text-field": "^1.0.0",
>     "vaadin-themable-mixin": "vaadin/vaadin-themable-mixin#^1.1.0",
>     "paper-icon-button": "^2.0.0"
> ```

Required by the demo, right? Should be a dev dependency then.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[bower.json, line 28 at r8](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-KofYhq0K8d5AY_nngml:-Kofjw7sk2oDCe4YlEoZ:b-896fix) ([raw file](https://github.com/vaadin/vaadin-combo-box/blob/c7ec7f3d61e27554ef03558d60f94ed74e3d9484/bower.json#L28)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Required by the demo, right? Should be a dev dependency then.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->

platosha commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r9.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468#-:-KofvJygJturBHZVMklP:bnfp4nl)*
<!-- Sent from Reviewable.io -->

platosha commented:
Reviewed 1 of 8 files at r1, 1 of 5 files at r2, 2 of 3 files at r3, 3 of 11 files at r6, 2 of 3 files at r7.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-combo-box/468)*
<!-- Sent from Reviewable.io -->
